lib/widgets/sample_viewer_page.dart

```diff
@@ -0,0 +1,185 @@
+
+/// A page that displays a list of sample categories.
+class SampleViewerPage extends StatefulWidget {
+  const SampleViewerPage({super.key, this.category, this.isSearchable = true});
+  final SampleCategory? category;
+  final bool isSearchable;
+
+  @override
+  State<SampleViewerPage> createState() => _SampleViewerPageState();
+}
+
+class _SampleViewerPageState extends State<SampleViewerPage> {
+  final _allSamples = <Sample>[];
+  final _searchFocusNode = FocusNode();
+  final _textEditingController = TextEditingController();
+  var _filteredSamples = <Sample>[];
+  var _ready = false;
+  var _searchHasFocus = false;
+
+  @override
+  void initState() {
+    super.initState();
+    loadSamples();
+    _searchFocusNode.addListener(
+      () {
+        if (_searchFocusNode.hasFocus != _searchHasFocus) {
+          setState(() => _searchHasFocus = _searchFocusNode.hasFocus);
+        }
+      },
+    );
+  }
+
+  @override
+  void dispose() {
+    _textEditingController.dispose();
+    _searchFocusNode.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('ArcGIS Maps SDK for Flutter Samples'),
+      ),
+      body: Column(
+        children: [
+          Visibility(
+            visible: widget.isSearchable,
+            child: Container(
+              padding: const EdgeInsets.symmetric(horizontal: 8),
+              child: TextField(
+                focusNode: _searchFocusNode,
+                controller: _textEditingController,
+                onChanged: onSearchChanged,
+                autocorrect: false,
+                decoration: InputDecoration(
+                  hintText: 'Search...',
+                  suffixIcon: IconButton(
+                    icon: Icon(
+                      _searchHasFocus ? Icons.cancel : Icons.search,
+                    ),
+                    onPressed: onSearchSuffixPressed,
+                  ),
+                ),
+              ),
+            ),
+          ),
+          if (_ready)
+            Expanded(
+              child: Listener(
+                onPointerDown: (_) =>
+                    FocusManager.instance.primaryFocus?.unfocus(),
+                child: SampleListView(samples: _filteredSamples),
+              ),
+            )
+          else
+            const Center(
+              child: Text('Loading samples...'),
+            ),
+        ],
+      ),
+    );
+  }
+
+  void onSearchSuffixPressed() {
+    if (_searchHasFocus) {
+      _textEditingController.clear();
+      FocusManager.instance.primaryFocus?.unfocus();
+      onSearchChanged('');
+    } else {
+      _searchFocusNode.requestFocus();
+    }
+  }
+
+  Future<void> loadSamples() async {
+    final jsonString =
+        await rootBundle.loadString('assets/generated_samples_list.json');
+    final sampleData = jsonDecode(jsonString);
+    for (final s in sampleData.entries) {
```

Comment:
One of the new lint rules is triggered by this line. To fix the lint warning, we have to call `jsonDecode` like this:

```dart
    final sampleData = jsonDecode(jsonString) as Map<String, dynamic>;
```